### ISSUE TITLE
Inject multiple implementations of an interface with Type based injection

### ISSUE THREAD

**Issue by stortiz-lifeworks** [kodemore/kink#13]
A very useful feature when using Type based injection would be the ability to inject all the implementations for a given interface as a list.

Here is an example of this same feature from the Java Spring documentation:
https://docs.spring.io/spring-framework/docs/4.3.12.RELEASE/spring-framework-reference/htmlsingle/#beans-autowired-annotation
![Screenshot 2021-06-09 at 15 03 56](https://user-images.githubusercontent.com/83952589/121369731-0425ca80-c934-11eb-8dfc-022fe4469ab2.png)

The way it would be used would be as follows:

```Python
class IMyService:
    def my_function(self, val: str) -> str:
        ...


@inject(alias=IMyService)
class FirstService(IMyService):
    def my_function(self, val: str) -> str:
        return f'First {val}'


@inject(alias=IMyService)
class SecondService(IMyService):
    def my_function(self, val: str) -> str:
        return f'Second {val}'

@inject
def list_vals(services: List[IMyService]) -> List[str]:
    return [service.my_function() for service in services]
```

In the snippet above, the parameter ```services``` of ```list_vals``` would receive the objects of both classes implementing ```IMyService```.

**Comment by dkraczkowski** [comment]
@stortiz-lifeworks Have a look here: https://github.com/kodemore/kink/pull/14 
Had to drop python3.6 support as I am short on time and actually lifecycle of 3.6 ends in december so I dont think I care anymore about such an old version.

**Comment by stortiz-lifeworks** [comment]
I have tested it and that is the requested issue.

But in the example added to the documentation there is a "register=False" argument passed to the decorator that is raising an exception when used, and that does not appear in the documentation.

Thanks.

**Comment by dkraczkowski** [comment]
@stortiz-lifeworks Available in 0.6.0 - https://github.com/kodemore/kink/actions/runs/935773769
Closing the issue now.
